semantic-conventions/syntax.md

```diff
@@ -0,0 +1,235 @@
+# Semantic Convention YAML Language
```

Comment:
As stated in the PR description, this file is copied from https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/syntax.md so that we can have any changes that affect the input format of the semantic convention generator in one PR in one repo, instead of split over spec and build-tools.